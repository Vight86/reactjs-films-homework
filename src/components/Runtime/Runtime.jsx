import React from 'react';
import propTypes from 'prop-types';
import styles from './Runtime.scss';

const Runtime = ({ children }) => {
  let hour = Math.floor(children / 60);
  hour = hour !== 0 ? `${hour}h` : '';
  let min = children % 60;
  min = min !== 0 ? `${min}m` : '';

  return (
    <span className={styles.runtime}>{`${hour} ${min}`.trim()}</span>
  );
};

Runtime.propTypes = {
  children: propTypes.number.isRequired,
};

export default Runtime;
