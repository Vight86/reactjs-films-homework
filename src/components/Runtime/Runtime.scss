@import '../../utils/variables';

.runtime {
  color: $color-light;
  font-size: 2.4em;
  font-weight: normal;
  letter-spacing: 0.04em;
}
