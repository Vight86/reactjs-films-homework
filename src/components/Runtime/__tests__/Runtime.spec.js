import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Runtime from '../index';

const setUp = (children) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Runtime>{children}</Runtime>);
  return renderer.getRenderOutput();
};

describe('render Runtime component', () => {
  it('render without props', () => {
    const result = setUp();
    expect(result).toMatchSnapshot();
  });

  it('render with props', () => {
    const result = setUp(123);
    expect(result).toMatchSnapshot();
  });
});
