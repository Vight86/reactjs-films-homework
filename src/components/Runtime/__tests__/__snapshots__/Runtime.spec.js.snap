// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Runtime component render with props 1`] = `
<span
  className="runtime"
>
  2h 3m
</span>
`;

exports[`render Runtime component render without props 1`] = `
<span
  className="runtime"
>
  
</span>
`;
